material-ui/material-ui.d.ts

```diff
@@ -1115,13 +1120,13 @@ declare namespace __MaterialUI {
         export class TableBody extends React.Component<TableBodyProps, {}> {
         }
 
-        interface TableFooterProps extends React.Props<TableFooter> {
+        interface TableFooterProps extends HasStyleAndClassNameProps<TableFooter> {
```

Comment:
Based on the `render()` method in the material-ui source code, since `tfoot` is passed all `other`  properties, use `React.HTMLAttributesBase<TableFooter>` instead of `HasStyleAndClassNameProps`.